canonical-web-and-design/maas-ui
# feat(ui): add action/selector/reducer for setting active machines.

## Done

- Added action/selectors/reducers for setting active machines. Setting a machine as active lets the server know to send all the data of that machine whenever it is updated, otherwise it only sends the data required for the machine list.
- Updated machine reducer tests to use the test factories

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to a machine details page and check that `machine.active` is set to the machine's system_id.
- Perform an action on the machine (any should do) and check that the machine is updated with all of the details data intact.
- Navigate back to the machine list and check that `machine.active` is set back to `null`.

## Fixes

Fixes #1820
